### ISSUE TITLE
[Feature] Add support for commands in Redis plugins

### ISSUE THREAD

**Issue by nidhi-nair** [appsmithorg/appsmith#2941]
Feature request to modify the query pane of the Redis plugin to support a drop down for all possible commands along with another field that only takes in arguments. This experience will be similar to the Firestore plugin.

**Comment by mohanarpit** [comment]
@nidhi-nair Wouldn't it be easier for a user to just type in a Redis command as they would in `redis-cli`? This way users can leverage the power of Redis without any intervention by the Appsmith team. 
With a dropdown, I feel that users will be limited by the set of commands that we expose.

**Comment by nidhi-nair** [comment]
True, the UX for redis is the CLI. The break is between other plugins in Appsmith where everything that one can do is visible I suppose. Does autocomplete for commands sound like an easier to understand solution? @Nikhil-Nandagopal , thoughts?

**Comment by Nikhil-Nandagopal** [comment]
Looking at the Redis docs, I thought we would be able to cover every single use case via UI. It just felt very simple to use like in firestore

**Comment by mohanarpit** [comment]
@Nikhil-Nandagopal Sorry, I don't understand. Are you in favor of a CLI interface? Or a dropdown interface? The number of Redis commands is large; with each command having its own set of parameters. Ref: [https://redis.io/commands](https://redis.io/commands). I think there's diminishing returns of exposing these commands via Dropdown with checks on the parameters themselves. 

If anything, I would upvote the autocomplete as suggested by @nidhi-nair. We can do this for SQL, Redis and any other plugin with a CLI interface.

**Comment by Nikhil-Nandagopal** [comment]
@mohanarpit auto-complete still needs you to learn the syntax. Dropdowns are self-explanatory but I am fine pushing this for later

**Comment by mohanarpit** [comment]
I vote for doing this later. The advantage with the CLI is that there is a world of documentation informing folks how to do this. With dropdown-like interfaces, the user is dependent on us to publish documentation educating them about this. 

Hence, where possible, I would always lean towards a CLI experience; thereby exposing the raw power of the plugin.

**Comment by trishaanand** [comment]
NOT a Good First Issue.
